src/DeploymentProvider/AzureSpringCloudDeploymentProvider.ts

```diff
@@ -0,0 +1,207 @@
+        let filteredResources: Array<Models.ServiceResource> = [];
+        serviceList.forEach(service => {
+            if(service.name == this.params.ServiceName) {
+                filteredResources.push(service);
+            }
+        });
+        if (!filteredResources || filteredResources.length == 0) {
+            throw new Error('ResourceDoesntExist' + this.params.ServiceName);
+        }
+        else if (filteredResources.length == 1) {
+            const beginStr = '/resourceGroups/';
+            const endStr = '/providers/Microsoft.AppPlatform/Spring'
+            const beginIndex = filteredResources[0].id.indexOf(beginStr) + beginStr.length;
+            const endIndex = filteredResources[0].id.indexOf(endStr);
+            if (beginIndex == -1 || endIndex == -1 || beginIndex >= endIndex) {
+                core.debug('ResourceGroupNameParseErrorWithId:' + filteredResources[0].id);
+                throw new Error('ResourceGroupNameParseError');
+            }
+            this.params.ResourceGroupName = filteredResources[0].id.substring(beginIndex, endIndex);
+            core.debug("resource group name: " + this.params.ResourceGroupName);
+        }
+        else { //Should never ever ever happen
+            throw new Error('DuplicateAzureSpringCloudName');
+        }
+        const serviceResponse = await this.client.services.get(this.params.ResourceGroupName, this.params.ServiceName);
+        core.debug("service response: " + serviceResponse._response.bodyAsText);
+    }
+
+    public async DeployAppStep() {
+        switch (this.params.Action) {
+
+            case Actions.deploy: {
+                await this.performDeployAction();
+                break;
+            }
+
+            case Actions.setProduction: {
+                await this.performSetProductionAction();
+                break;
+            }
+
+            case Actions.deleteStagingDeployment: {
+                await this.performDeleteStagingDeploymentAction();
+                break;
+            }
+
+            default:
+                throw Error('UnknownOrUnsupportedAction' + this.params.Action);
+        }
+    }
+
+    private async performDeleteStagingDeploymentAction() {
+        console.log('Delete staging deployment action');
+        const deploymentName = await dh.getStagingDeploymentName(this.client, this.params);
+        this.params.DeploymentName = deploymentName;
+        console.log(`Action for service ${this.params.ServiceName} app ${this.params.AppName} deployment ${deploymentName}`);
+        if (deploymentName) {
+            await dh.deleteDeployment(this.client, this.params);
+        } else {
+            throw Error('NoStagingDeploymentFound');
+        }
+        console.log('Delete staging deployment action successful');
+        return deploymentName;
+    }
+
+    private async performSetProductionAction() {
+        console.log('Set production action');
```

Comment:
Probably remove this? It may be better to log all necessary information in a line, instead of breaking them into multiple lines which makes the log difficult to read especially when tasks are being executed in parallel.

Reply:
I think it is better to give the info in one line. However, I cannot know whether the action can pass the validation. So if I remove this line, I need to add the action info in each Error and each are duplicate action info. I can get the app and deployment name after the validation. I am not sure which is a better implementation.